lib/attr_encrypted.rb

```diff
@@ -341,7 +341,8 @@ def load_iv_for_attribute(attribute, algorithm)
 
       def load_salt_for_attribute(attribute)
         encrypted_attribute_name = self.class.encrypted_attributes[attribute.to_sym][:attribute]
-        salt = send("#{encrypted_attribute_name}_salt") || send("#{encrypted_attribute_name}_salt=", SecureRandom.hex)
+        salt = send("#{encrypted_attribute_name}_salt") || send("#{encrypted_attribute_name}_salt=", [SecureRandom.random_bytes].pack("m"))
+        salt = salt.size > 16 ? salt.unpack("m").first : salt
```

Comment:
@pauljm the usage of SecureRandom.hex was just introduced the other day. As such, I assumed that it was never used; previously generated salt should have been only 16 bytes. However, you are correct, any newly generated salt using SecureRadom.hex would have a size of 32. 

I like your suggestion as it's the safest, namely, to prepend a non hex char and use it as the identifier. I'll roll with that. 

Thank you for your time.